app/Http/Requests/Album/ArchiveAlbumsRequest.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+namespace App\Http\Requests\Album;
+
+use App\Contracts\AbstractAlbum;
+use App\Http\Requests\BaseApiRequest;
+use App\Http\Requests\Contracts\HasAlbums;
+use App\Http\Requests\Traits\HasAlbumsTrait;
+use App\Rules\AlbumIDListRule;
+
+/**
+ * @implements HasAlbums<AbstractAlbum>
+ */
+class ArchiveAlbumsRequest extends BaseApiRequest implements HasAlbums
+{
+	use HasAlbumsTrait;
+
+	/**
+	 * {@inheritDoc}
+	 */
+	public function authorize(): bool
+	{
+		return $this->authorizeAlbumsAccess($this->albums);
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	public function rules(): array
+	{
+		return [
+			HasAlbums::ALBUM_IDS_ATTRIBUTE => ['required', new AlbumIDListRule()],
+		];
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	protected function processValidatedValues(array $values, array $files): void
+	{
+		// TODO: `App\Actions\Album\Archive::compressAlbum` iterates over the original size variant of each photo in the album; we should eagerly load them for higher efficiency.
+		$this->albums = $this->albumFactory->findAbstractAlbumsOrFail(
+			explode(',', $values[HasAlbums::ALBUM_IDS_ATTRIBUTE])
```

Comment:
Because, the browser needs to download the final zip file and ask the user where to store it.

Hence, the frontend code redirects the browser to a "normal" URL and the browser triggers an ordinary GET-request by itself. As the GET request must not have any body, the IDs must be part of the URL and the simplest way to encode all the IDs in the URL is as a comma-separated string. See here:

See here: https://github.com/LycheeOrg/Lychee-front/blob/e712c60761d6bcc5428e34f729f2974262fa04fe/scripts/main/album.js#L1083-L1089

Same holds for archiving of photos.

I would be more than happy to have proper JSON request everywhere. If you know a way how to make an AJAX request by the frontend, but don't parse the response programmatically, but let the browser "download" it (i.e. store it in a file), I am all yours.